addons/web_editor/controllers/main.py

```diff
@@ -519,7 +520,16 @@ def shape(self, module, filename, **kwargs):
         """
         svg = None
         if module == 'illustration':
-            attachment = request.env['ir.attachment'].sudo().search([('url', '=like', request.httprequest.path), ('public', '=', True)], limit=1)
+            attachment = request.env['ir.attachment'].sudo().search(
+                [
+                    ('public', '=', True),
+                    ('type', '=', 'binary'),
+                    '|',
+                    ('url', '=like', request.httprequest.path + '?%'),
+                    ('url', '=', request.httprequest.path),  # old URLs have no parameters
+                ],
+                order='create_date desc',
+                limit=1)
```

Comment:
@bso-odoo Actually I just discussed this with @JKE-be. Since the path of the illustration contains its slug, we can simply use `unslug(filename)` to get the ID of the attachment, and then we just need to browse and check that it's binary and public, this will also be much faster than a string search with a wildcard.